packages/ra-ui-materialui/src/input/AutocompleteInput/index.js

```diff
@@ -0,0 +1,396 @@
+import get from 'lodash/get';
+import { makeStyles } from '@material-ui/core/styles';
+import Downshift from 'downshift';
+import { useTranslate, useInput } from 'ra-core';
+
+import AutocompleteInputTextField from './AutocompleteInputTextField';
+import AutocompleteSuggestionList from './AutocompleteSuggestionList';
+import getSuggestions from './getSuggestions';
+import { InputHelperText } from '..';
+
+const useStyles = makeStyles({
+    container: {
+        flexGrow: 1,
+        position: 'relative',
+    },
+    root: {},
+    suggestionsContainer: {
+        zIndex: 2,
+    },
+    suggestionsPaper: {
+        maxHeight: '50vh',
+        overflowY: 'auto',
+    },
+    suggestionsList: {
+        margin: 0,
+        padding: 0,
+        listStyleType: 'none',
+    },
+});
+
+/**
+ * An Input component for an autocomplete field, using an array of objects for the options
+ *
+ * Pass possible options as an array of objects in the 'choices' attribute.
+ *
+ * By default, the options are built from:
+ *  - the 'id' property as the option value,
+ *  - the 'name' property an the option text
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'Male' },
+ *    { id: 'F', name: 'Female' },
+ * ];
+ * <AutocompleteInput source="gender" choices={choices} />
+ *
+ * You can also customize the properties to use for the option name and value,
+ * thanks to the 'optionText' and 'optionValue' attributes.
+ * @example
+ * const choices = [
+ *    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
+ *    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
+ * ];
+ * <AutocompleteInput source="author_id" choices={choices} optionText="full_name" optionValue="_id" />
+ *
+ * `optionText` also accepts a function, so you can shape the option text at will:
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
+ * <AutocompleteInput source="author_id" choices={choices} optionText={optionRenderer} />
+ *
+ * The choices are translated by default, so you can use translation identifiers as choices:
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'myroot.gender.male' },
+ *    { id: 'F', name: 'myroot.gender.female' },
+ * ];
+ *
+ * However, in some cases (e.g. inside a `<ReferenceInput>`), you may not want
+ * the choice to be translated. In that case, set the `translateChoice` prop to false.
+ * @example
+ * <AutocompleteInput source="gender" choices={choices} translateChoice={false}/>
+ *
+ * The object passed as `options` props is passed to the material-ui <AutoComplete> component
+ *
+ * @example
+ * <AutocompleteInput source="author_id" options={{ fullWidth: true }} />
+ */
+const AutocompleteInput = ({
+    classes: classesOverride,
+    allowEmpty,
+    choices,
+    fullWidth,
+    helperText,
+    label,
+    limitChoicesToValue,
```

Comment:
this one isn't documented and I don't know what it means

Reply:
It is in the current version: https://marmelab.com/react-admin/Inputs.html#autocompleteinput

> If you want to limit the initial choices shown to the current value only, you can set the limitChoicesToValue prop.